entity-service/src/main/kotlin/com/egm/stellio/entity/repository/Neo4jRepository.kt

```diff
@@ -18,33 +17,45 @@ import java.time.Instant
 import java.time.ZoneOffset
 import javax.annotation.PostConstruct
 
+sealed class SubjectNodeInfo(val id: String, val label: String)
+class EntitySubjectNode(id: String) : SubjectNodeInfo(id, "Entity")
+class AttributeSubjectNode(id: String) : SubjectNodeInfo(id, "Attribute")

```

Comment:
Why not push it further ? RelationshipSubjectNode / PropertySubjectNode ? If I understand correctly the more label you put the faster it goes ?

Reply:
Because there is currently no use-case for such specializations. Other queries known on which labels they apply and don't need to be parameterized on label name.